src/main/scala/com/fulcrumgenomics/fastq/DemuxFastqs.scala

```diff
@@ -238,33 +264,52 @@ val qualityFormat: Option[FastqQualityFormat] = None,
- @arg(doc="Date the run was produced, to insert into the read group header") val runDate: Option[Iso8601Date] = None
+ @arg(doc="Date the run was produced, to insert into the read group header") val runDate: Option[Iso8601Date] = None,
+ @arg(doc="Output FASTQs (.fastq.gz) in addition to BAM files") val outputFastqs: Boolean = false
 ) extends FgBioTool with LazyLogging {
 
   import DemuxFastqs._
 
+  private[fastq] val metricsPath = metrics.getOrElse(output.resolve(DefaultDemuxMetricsFileName))
+
   validate(inputs.length == readStructures.length, "The same number of read structures should be given as FASTQs.")
   validate(readStructures.flatMap(_.sampleBarcode).nonEmpty, s"No sample barcodes found in read structures: " + readStructures.map(_.toString).mkString(", "))
 
-  /** True if the read structure implies paired end reads will be produced, false otherwise. */
-  val pairedEnd: Boolean = readStructures.count(_.template.nonEmpty) match {
+  private val pairedEnd = readStructures.count(_.template.nonEmpty) match {
     case 1 => false
     case 2 => true
     case n => invalid(s"Found $n read structures with template bases but expected 1 or 2.")
   }
 
   Io.assertReadable(inputs)
-  Io.assertReadable(sampleSheet)
+  Io.assertReadable(metadata)
   Io.assertWritableDirectory(output)
 
+  /** The quality format converter. */
+  private val solexaQualityConverter: SolexaQualityConverter = SolexaQualityConverter.getSingleton
+
+  /** Create a sample sheet from either the input sample sheet path or the metadata CSV. */
+  private val sampleSheet: SampleSheet = {
+    val lines = Io.readLines(metadata).toSeq
+    if (lines.exists(_.contains("[Data]"))) {
+      logger.info("Assuming input metadata file is an Illumina Experiment Manager Sample Sheet.")
+      SampleSheet(lines.toIterator, lane=None)
+    }
+    else {
+      logger.info("Assuming input metadata file is simple CSV file.")
+      SampleSheet(Iterator("[Data]") ++ lines, lane=None)
+    }
+  }
+
   override def execute(): Unit = {
     // Get the FASTQ quality encoding format
     val qualityFormat = this.qualityFormat.getOrElse(determineQualityFormat(inputs, this.qualityFormat, Some(this.logger)))
 
     // Read in the sample sheet and create the sample information
-    val samplesFromSampleSheet = SampleSheet(this.sampleSheet).map(s => withCustomSampleBarcode(s, columnForSampleBarcode))
+    val samplesFromSampleSheet = sampleSheet.map(s => withCustomSampleBarcode(s, columnForSampleBarcode))
     val samples                = samplesFromSampleSheet.toSeq :+ unmatchedSample(samplesFromSampleSheet.size, this.readStructures)
-    val sampleInfos            = samples.map(toSampleInfo)
+    val sampleInfos            = samples.map(sample => SampleInfo(sample, sample.sampleName == UnmatchedSampleId))
+    val sampleToWriter         = sampleInfos.map { info =>  (info.sample, toWriter(info)) }.toMap
```

Comment:
The two are equivelant, but I kinda prefer this syntax for map creation, how about you:
`sampleInfos.map { info =>  info.sample -> toWriter(info) }.toMap`